Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift

```diff
@@ -81,4 +81,12 @@ extension StorageCategory: StorageCategoryBehavior {
                      resultListener: StorageListOperation.ResultListener?) -> StorageListOperation {
         return plugin.list(options: options, resultListener: resultListener)
     }
+
+    // MARK: - Async API -
```

Comment:
[suggestion] since I've done this for API and DataStore and List APIs.. 
- It didn't seem too much work to all all async flavours in one single PR, see API https://github.com/aws-amplify/amplify-ios/pull/2140/files and DataStore https://github.com/aws-amplify/amplify-ios/pull/2126 Up to you to add ontop of this PR or do another one later
- I added them undernearth each corresponding previous API, instead of creating a new section below them. just pointing it out, up to your personal preference. Might be easier to spot the API differences and easier to remove previous API later and use the same API documentation (if it exists)

Reply:
looks like this PR has more unit testing and better documentation, will retract my suggestion above